jan2025/helpers.js

```diff
@@ -0,0 +1,369 @@
+      }
+    }
+
+    // Filter flagset category change
+    if (nodeData?.["type"] === 500 && nodeData?.["data"]?.["category"] === "Listed building consent") {
+      newFlowData[nodeId]["data"]["category"] = "Works to listed buildings";
+      logs += `${timestamp} Updated Filter category (node ${nodeId}); `;
+    }
+
+    // Article 4s (options and selectable planning constraints)
+    if (nodeData?.["type"] === 200 && nodeData?.["data"]?.["val"]?.startsWith("article4")) {
+      newFlowData[nodeId]["data"]["val"] = nodeData["data"]["val"].replace("article4", "articleFour");
+      logs += `${timestamp} Updated article4 option val (node ${nodeId}); `;
+    }
+
+    if (
+      nodeData?.["type"] === 11 &&
+      (nodeData?.["data"]?.["dataValues"]?.includes("article4") || nodeData?.["data"]?.["dataValues"]?.includes("article4.caz"))
+    ) {
+      newFlowData[nodeId]["data"]["dataValues"] = nodeData["data"]["dataValues"].map(
+        current => current.startsWith("article4") ? current.replace("article4", "articleFour") : current
+      );
+      logs += `${timestamp} Updated planning constraints selectable data values (node ${nodeId}); `;
+    }
+
+    // Listed building grades (options only)
+    if (nodeData?.["type"] === 200 && nodeData?.["data"]?.["val"]?.startsWith("listed.")) {
+      const current = nodeData["data"]["val"];
+      newFlowData[nodeId]["data"]["val"] = listedGradeChanges[current];
+      logs += `${timestamp} Updated listed building grade option val (node ${nodeId}); `;
+    }
+
+    // Flood risk zones (options only)
+    if (nodeData?.["type"] === 200 && nodeData?.["data"]?.["val"]?.startsWith("flood.")) {
+      const current = nodeData["data"]["val"];
+      newFlowData[nodeId]["data"]["val"] = floodZoneChanges[current];
+      logs += `${timestamp} Updated flood risk zone option val (node ${nodeId}); `;
+    }
+
+    // Property types (option or setvalue type)
+    if (
+      nodeData?.["data"]?.["val"] && 
+      Object.keys(propertyTypeChanges).includes(nodeData["data"]["val"])
+    ) {
+      const current = nodeData["data"]["val"];
+      newFlowData[nodeId]["data"]["val"] = propertyTypeChanges[current];
+      logs += `${timestamp} Updated property type value (node ${nodeId}); `
+    }
+
+    // Lingering draw boundary props in calculate formulas and defaults
+    if (nodeData?.["type"] === 700) {
+      Object.keys(nodeData?.["data"]?.["defaults"] || {})?.map((current) => {
+        if (Object.keys(drawBoundaryChanges).includes(current)) {
+          const proposed = drawBoundaryChanges[current];
+          newFlowData[nodeId]["data"]["defaults"][proposed] = nodeData["data"]["defaults"][current];
+          delete newFlowData[nodeId]["data"]["defaults"][current];
+          newFlowData[nodeId]["data"]["formula"] = nodeData["data"]["formula"].replaceAll(current, proposed);
+          logs += `${timestamp} Updated Calculate formula and defaults (node ${nodeId}); `;
+        }
+      });
```

Comment:
Great catch here - we'll need to bear this in mind of all future data value migrations.